src/chainstate/stacks/address.rs

```diff
@@ -92,6 +94,49 @@ pub enum PoxAddress {
     Addr32(bool, PoxAddressType32, [u8; 32]),
 }
 
+/// Serializes a PoxAddress as a B58 check encoded address or a bech32 address
+pub fn pox_addr_b58_serialize<S: Serializer>(
```

Comment:
These methods don't preserve the `AddressHashMode`, which makes them unsuitable for use within the sortition DB.

I think you have two options:
* You can update the serializers for `PoxAddress` so that the relevant metadata is preserved, or
* You can create a JSON-specific serialization method that will be used for the event stream.

I'm more in favor of the latter since, selfishly, I'm in the process of using already-spun-up nodes for testing and don't want to have to sync from genesis again, but that's something I can work around on my end.  Up to you how you want to proceed.  If you take the latter, please document it.

Reply:
Just implemented the latter!

Reply:
I added docs for the serialization in the event dispatcher doc already.